src/main/java/com/_up/megastore/services/implementations/UserService.java

```diff
@@ -0,0 +1,53 @@
+package com._up.megastore.services.implementations;
+
+import com._up.megastore.controllers.requests.CreateUserRequest;
+import com._up.megastore.controllers.responses.UserResponse;
+import com._up.megastore.data.model.User;
+import com._up.megastore.data.repositories.IUserRepository;
+import com._up.megastore.services.interfaces.IUserService;
+import com._up.megastore.services.mappers.UserMapper;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.stereotype.Service;
+
+@Service
+public class UserService implements IUserService {
+
+  private final IUserRepository userRepository;
+  private final PasswordEncoder passwordEncoder;
+
+  public UserService(IUserRepository userRepository, PasswordEncoder passwordEncoder) {
+    this.passwordEncoder = passwordEncoder;
+    this.userRepository = userRepository;
+  }
+
+  @Override
+  public UserResponse saveUser(CreateUserRequest createUserRequest) {
+    validateRequest(createUserRequest);
+    User user = createUser(createUserRequest);
+    return UserMapper.toUserResponse(userRepository.save(user));
+  }
+
+  private void validateRequest(CreateUserRequest createUserRequest) {
```

Comment:
Muy quisquilloso pero usaría un nombre más específico si después vamos a poder actualizar usuarios
```suggestion
  private void validateCreateUserRequest(CreateUserRequest createUserRequest) {
```